static_frame/test/unit/test_index.py

```diff
@@ -1648,6 +1650,50 @@ def test_index_display_a(self) -> None:
         idx.append('e')
         post = idx.display(DisplayConfig(type_show=False, type_color=False))
         self.assertEqual(str(post), 'a\nb\nc\nd\ne')
+    
+    #---------------------------------------------------------------------------
+    def test_sizeof_simple_index(self):
+        # TODO: Look into determistically finding the size on different platforms / versions of python
+        idx = Index(('a', 'b', 'c'))
+        self.assertEqual(getsizeof(idx), 850)
+    
+    def test_sizeof_object_index(self):
+        idx = Index((1, 'b', (2, 3)))
+        self.assertEqual(getsizeof(idx), 846)
+    
+    def test_sizeof_loc_is_iloc(self):
+        idx = Index((0, 1, 2), loc_is_iloc=True)
+        self.assertEqual(getsizeof(idx), 272)
+    
+    def test_sizeof_empty_index(self):
+        idx = Index(())
+        self.assertEqual(getsizeof(idx), 592)
+    
+    def test_sizeof_name_adds_size(self):
+        idx1 = Index(('a', 'b', 'c'))
+        idx2 = idx1.rename('with_name')
+        self.assertTrue(getsizeof(idx1) < getsizeof(idx2))
+    
+    def test_sizeof_more_values_adds_size(self):
+        idx1 = Index(('a', 'b', 'c'))
+        idx2 = Index(('a', 'b', 'c', 'd'))
+        self.assertTrue(getsizeof(idx1) < getsizeof(idx2))
+
+    def test_sizeof_more_nested_values_adds_size(self):
+        idx1 = Index((1, 'b', (2, 3)))
+        idx2 = Index((1, 'b', (2, 3, 4, 5)))
+        self.assertTrue(getsizeof(idx1) < getsizeof(idx2))
+    
+    def test_sizeof_more_doubly_nested_values_maintains_size(self):
```

Comment:
This is how it works now. I'm considering changing total_getsizeof to count the size of nested containers.

Reply:
To do this, I'm going to recreate the "handlers" concept from the [total_sizeof recipe](https://code.activestate.com/recipes/577504/) rather than checking for instances of iterables within the containers.

This way, iterables that represent the full contents in __sizeof__ (like numpy arrays) can be handled differently than those that do not (like lists of lists).

Reply:
Doubly-nested values no longer maintain size.